src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -283,12 +299,29 @@ def set_key(cmd,
             raise CLIError(str(exception))
 
         if retrieved_kv is None:
+            if content_type and __is_json_content_type(content_type):
+                try:
+                    # Ensure that provided value is valid JSON. Error out if value is invalid JSON.
+                    value = '""' if value is None else value
+                    value = json.dumps(json.loads(value), separators=(',', ':'))
+                except ValueError:
```

Comment:
Why we need to convert string to json then back to string? loads() can be used to verify the json format but dumps() with separators=(',', ':') will change the value content by trimming out whitespaces.